cpp/ruapu.hpp

```diff
@@ -0,0 +1,39 @@
+#include <algorithm>
+#include <iostream>
+#include <string>
+#include <vector>
+
+#define RUAPU_IMPLEMENTATION
+#include "../ruapu.h"
+
+class ruapu
+{
+public:
+    ruapu() { 
+      static int initialised = 0;
+      if (!initialised) {
+        ruapu_init();
+        initialised = 1;
+      }
+    }
```

Comment:
Make this a static class method.

```suggestion
    static void init() { 
      static int initialised = 0;
      if (!initialised) {
        ruapu_init();
        initialised = 1;
      }
    }
```